### PR TITLE
Implement min/max for Index/MultiIndex

### PR DESCRIPTION
**Pull request by itholic**
Implement min/max for Index/MultiIndex

For Index

```python
>>> idx = ks.Index([3, 2, 1])
>>> idx.min()
1
>>> idx.max()
3
```

For MultiIndex

```python
>>> idx = ks.MultiIndex.from_tuples([('a', 'x', 1), ('b', 'y', 2)])
>>> idx.min()
('a', 'x', 1)
>>> idx.max()
('b', 'y', 2)
```

### CODE REVIEW

File: databricks/koalas/indexes.py [hunk 1]
```diff
@@ -711,6 +711,80 @@ def symmetric_difference(self, other, result_name=None, sort=None):
...
+
+        For a MultiIndex, the maximum is determined lexicographically.
+
+        >>> idx = ks.MultiIndex.from_tuples([('a', 'x', 1), ('b', 'y', 2)])
+        >>> idx.min()
+        ('a', 'x', 1)
+        """
+        sdf = self._internal.sdf
+        min_row = sdf.select(F.min(F.struct(self._internal.index_scols))).head()
+        result = tuple(min_row[0][level] for level in range(self.nlevels))
```
Comment (ueshin) [inline databricks/koalas/indexes.py]: I guess `tuple(min_row[0])` should work.

File: databricks/koalas/indexes.py [hunk 2]
```diff
@@ -711,6 +711,80 @@ def symmetric_difference(self, other, result_name=None, sort=None):
...
+
+        For a MultiIndex, the maximum is determined lexicographically.
+
+        >>> idx = ks.MultiIndex.from_tuples([('a', 'x', 1), ('b', 'y', 2)])
+        >>> idx.max()
+        ('b', 'y', 2)
+        """
+        sdf = self._internal.sdf
+        max_row = sdf.select(F.max(F.struct(self._internal.index_scols))).head()
+        result = tuple(max_row[0][level] for level in range(self.nlevels))
```
Comment (ueshin) [inline databricks/koalas/indexes.py]: ditto.

File: docs/source/reference/indexing.rst [hunk 3]
```diff
@@ -40,6 +40,8 @@ Modifying and computations
    Index.is_interval
    Index.is_numeric
    Index.is_object
+   Index.min
+   Index.max
```
Comment (HyukjinKwon) [inline docs/source/reference/indexing.rst]: @itholic can you create another PR to add all missing functions in `MultiIndex` below?

Comment (itholic) [reply docs/source/reference/indexing.rst]: @HyukjinKwon okay, i'll take it

**Pull request merged**